### ISSUE TITLE
Allow modification of default renderers

### ISSUE THREAD

**Issue by cwrau** [klogging/klogging#145]
I've been happily using `RENDER_ANSI`, there is just one little detail I'd like to change.

I want a full ISO-8601 timestamp for the log messages;

```log
2023-07-27T10:40:31.540076647+02300  INFO [                main] :        Configuration : Reading configuration from klogging.json on the classpath
```

The rest of the log message is ok, but as it stands now I'd have to copy the whole code of the `RENDER_ANSI` to change a single field

**Comment by mjstrasser** [comment]
I am aware that it customer renderers are a bit tricky to implement. Possible with configuration DSL but not with JSON files.

The `RENDER_ANSI` renderer is modelled on a Log4J appender. It is designed for human reading during interactive execution with colouring. It omits the date for brevity.

The similar `RENDER_SIMPLE` renderer includes the date but omits the colours.

I have created an experimental `RENDER_ISO8601` renderer that uses ISO8601 timestamps and no colours. You can try it with the snapshot build that includes it.

**Comment by mjstrasser** [comment]
I believe the changes in #211 (i.e. changing `RenderString` and `EventSender` from functional types to interfaces) provide this need.

**Comment by cwrau** [comment]
> I have created an experimental `RENDER_ISO8601` renderer that uses ISO8601 timestamps and no colours. You can try it in version 0.4.21 and later.

Yes, that's been working nicely except for the missing colors 😁

> I believe the changes in #211 (i.e. changing `RenderString` and `EventSender` from functional types to interfaces) provide this need.

I don't know how that helps me not to copy the whole block? As far as I can see it, I'd have to copy the following block;

```kotlin
public val RENDER_ANSI: RenderString = object : RenderString {
    override fun invoke(event: LogEvent): String {
        val message = "${event.timestamp.localTime} ${event.level.colour5} [${event.context?.right20}] :" +
                " ${event.logger.right20} : ${event.evalTemplate()}"
        val maybeItems = if (event.items.isNotEmpty()) " : ${event.items}" else ""
        val maybeStackTrace = if (event.stackTrace != null) "\n${event.stackTrace}" else ""
        return message + maybeItems + maybeStackTrace
    }
}
```

just to remove the `.localTime` part

**Comment by mjstrasser** [comment]
@cwrau I have extracted console rendering of event items and stack trace, as you can see from the above commit.

That should make it simpler for you to create the renderer you want:
- Local time or ISO8601 timestamp
- Event level in colour or not
- Full context or truncated
- Full logger name or truncated
- With or without event items (if present)
- With or without stack trace (if present)

**Comment by cwrau** [comment]
> @cwrau I have extracted console rendering of event items and stack trace, as you can see from the above commit.
> 
> That should make it simpler for you to create the renderer you want:
> 
> * Local time or ISO8601 timestamp
> * Event level in colour or not
> * Full context or truncated
> * Full logger name or truncated
> * With or without event items (if present)
> * With or without stack trace (if present)

Sounds amazing!
I tried publishing it locally, but I always get `Cannot perform signing task ':jdk-platform-klogging:signJvmPublication' because it has no configured signatory`

I tried setting `SIGNING_KEY="$(gpg --armor --export-secret-keys 6ADCBD3FF206F341D2CF8E2C49D1B5ACCF53CA1C B)" SIGNING_KEY_ID=6ADCBD3FF206F341D2CF8E2C49D1B5ACCF53CA1C`, but that didn't help 😅

**Comment by mjstrasser** [comment]
Where were you trying to publish the library? 

I have included the new changes in version 0.7.1 of Klogging. If it is not available from Maven Central yet, you can specify:

```kotlin
repositories {
    // ...
    maven {
        url = uri("https://s01.oss.sonatype.org/content/repositories/releases/")
    }
}

dependencies {
    // ...
    implementation("io.klogging:klogging-jvm:0.7.1")
}
```

**Comment by cwrau** [comment]
> Where were you trying to publish the library?

Yes and no, I was trying to publish it locally, to test your changes, with `gradle publishToMavenLocal`

**Comment by chile12** [comment]
> I am aware that it customer renderers are a bit tricky to implement. Possible with configuration DSL but not with JSON files.

Just wanted to point out that you can very well point out either your own event sink or own renderer with the JSON config:
```json
  "sinks": {
    "stdout": {
      "renderWith": "com.stuff.common.logging.StdOut",
      "sendTo": "STDOUT"
    },
    "logback": {
      "eventSender": "com.stuff.common.logging.LogbackSink"
    }
  }
```
Both methods not yet documented at klogging.io

**Comment by mjstrasser** [comment]
> Both methods not yet documented at klogging.io

Not strictly true: please see [Custom rendering and sending](https://klogging.io/docs/concepts/rendering-and-sending/#custom-rendering-and-sending)
